avl_tree/AvlTree.java

```diff
@@ -0,0 +1,271 @@
+import java.util.Scanner;
+import java.util.NoSuchElementException;
+import java.util.Random;
+
+public class AvlTree {
+
+	public static void main(String[] args) {
+		AVL avl = new AVL();
+		Random rand = new Random();
+		int inp = 10;
+		System.out.println(avl.isEmpty());
+		for (int i = 0; i < inp; i++) {

```

Comment:
All you have to do is add 3-4 nodes by yourself and test the `delNode()` by putting it in `try-catch` block with some invalid entry.